# 02 – 1. Experiment: Hyperparameter: Beam Size × Max Length

Dieses Notebook untersucht die Decoding-Hyperparameter des besten Baseline-Modells
(BL4: AV-HuBERT Cocktail, MCoRec-feingetunet) auf einem 5-Session-Subset des Dev-Sets.

### Untersuchte Parameter

| Parameter | Werte | Erklärung |
|-----------|-------|-----------|
| `beam_size` | 4, 8, 12 | Anzahl parallel verfolgter Decoder-Hypothesen. Größere Werte → bessere WER, mehr Rechenaufwand |
| `max_length` | 10, 15, 18, 20 s | Maximale Länge eines Audio-/Video-Segments. Kürzere Segmente → mehr Schnitte mitten im Satz → schlechtere WER |

### Ergebnis (Vorschau)

Beste Kombination: **`beam_size=12`, `max_length=20`** (WER: 0.4954, Joint Error: 0.3239)  
Knapp dahinter: `beam_size=8`, `max_length=20` – bei deutlich weniger Rechenaufwand.

Die vollständige Analyse findet sich in Abschnitt 9.

**Hinweis zum Bugfix:** Dieser Lauf wurde **vor dem Bugfix** in `segmentation.py` durchgeführt (`min_duration_off` las fälschlicherweise den Wert von `min_duration_on`). Das ist **gewollt**: Der Bugfix wurde erst nach Abschluss der LLM- und Hyperparameter-Experimente entdeckt. Da der Bugfix allein die WER zunächst verschlechterte, wurde erst in `02j_`/`02k_` die Kombination aus Bugfix + `min_duration`-Optimierung erarbeitet, die schließlich das beste Ergebnis lieferte.

## 1 – GPU-Verfügbarkeit prüfen

In [1]:
!nvidia-smi

Thu Jan 29 13:58:04 2026       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 550.54.15              Driver Version: 550.54.15      CUDA Version: 12.5     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA A100-SXM4-80GB          On  |   00000000:01:00.0 Off |                    0 |
| N/A   34C    P0             80W /  500W |    9073MiB /  81920MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+-----

## 2 – GPU-Auswahl

In [2]:
import os

# Physische GPU-Auswahl: hier GPU 2 (siehe nvidia-smi)
os.environ["CUDA_DEVICE_ORDER"] = "PCI_BUS_ID"
os.environ["CUDA_VISIBLE_DEVICES"] = "2"


## 3 – CUDA-Verifikation

Sicherstellen, dass PyTorch die korrekte GPU sieht, bevor rechenintensive Experimente starten.

In [3]:
import torch

In [4]:
print("CUDA devices:", torch.cuda.device_count())
print("Device 0 name (sollte A100 sein):", torch.cuda.get_device_name(0))

CUDA devices: 1
Device 0 name (sollte A100 sein): NVIDIA A100-SXM4-80GB


## 4 – Setup: Arbeitsverzeichnis & Imports

`pg_utils_experiments` ist ein projektinternes Hilfsmodul, das die Boilerplate-Logik
für Inference-Läufe und Evaluation kapselt, um Codeduplizierung über Experiment-Notebooks
hinweg zu vermeiden.

In [5]:
import os, sys
import pandas as pd

# Arbeitsverzeichnis auf Repo-Root setzen (Voraussetzung für alle relativen Pfade)
project_baseline_path = "/home/josch080/Projektgruppe/mcorec_baseline"
os.chdir(project_baseline_path)

# Repo-Root auch zum Python-Suchpfad hinzufügen, damit projektinterne Module importierbar sind
# Repo-Root in sys.path, damit projektinterne Module importierbar sind
if project_baseline_path not in sys.path:
    sys.path.append(project_baseline_path)
    
# Projektinterne Hilfsfunktionen
from script.pg_utils_experiments import run_inference_for_experiment, run_eval_and_log, append_eval_results_for_experiments

/home/josch080/Projektgruppe/mcorec_baseline/script/inference.py:22: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
  if not hasattr(np, "object"):


## 5 – Verfügbare Basis-Modelle

Alle vier Baseline-Modelle werden hier registriert, auch wenn in diesem Notebook
nur `avsr_cocktail_finetuned` (BL4) für die Hyperparameter-Suche verwendet wird.
Das erlaubt es, das Dict bei Bedarf einfach auf andere Modelle auszudehnen.

In [6]:
BASE_MODELS = {
    "auto_avsr": {
        "model_type": "auto_avsr",
        "chkpt": "model-bin/auto_avsr/avsr_trlrwlr2lrs3vox2avsp_base.pth",
    },
    "avsr_cocktail": {
        "model_type": "avsr_cocktail",
        "chkpt": "model-bin/avsr_cocktail",
    },
    "avsr_cocktail_finetuned": {
        "model_type": "avsr_cocktail",
        "chkpt": "model-bin/avsr_cocktail_mcorec_finetune",
    },
    "muavic_en": {
        "model_type": "muavic_en",
        "chkpt": "nguyenvulebinh/AV-HuBERT-MuAViC-en",
    },
}


## 6 – Session-Subset

Statt aller Dev-Sessions werden nur 5 zufällig ausgewählt, um die Laufzeit zu begrenzen.
Die Sessions wurden repräsentativ für das gesamte Dev-Set gewählt.

In [3]:
# 5-Session-Subset des Dev-Sets für schnellere Hyperparameter-Suche
SESSION_IDS = ["session_40", "session_43", "session_49", "session_50", "session_54"]

## 7 – Experiment-Grid: 3 × 4 Kombinationen

10 Experimente decken das Grid `beam_size ∈ {4, 8, 12}` × `max_length ∈ {15, 18, 20}` ab,
plus ein Kontroll-Setup mit `max_length=10`, das als untere Schranke dient.

Der Experiment-Name kodiert die Parameter direkt (z.B. `E09_bs12_len20`),
was die Output-Verzeichnisse selbsterklärend macht.

In [4]:
EXPERIMENTS = {
    # 15 Sekunden
    "E01_bs4_len15": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 4,
        "max_length": 15,
        "comment": "beam=4, len=15",
    },
    "E02_bs8_len15": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 8,
        "max_length": 15,
        "comment": "beam=8, len=15",
    },
    "E03_bs12_len15": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 12,
        "max_length": 15,
        "comment": "beam=12, len=15",
    },

    # 18 Sekunden
    "E04_bs4_len18": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 4,
        "max_length": 18,
        "comment": "beam=4, len=18",
    },
    "E05_bs8_len18": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 8,
        "max_length": 18,
        "comment": "beam=8, len=18",
    },
    "E06_bs12_len18": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 12,
        "max_length": 18,
        "comment": "beam=12, len=18",
    },

    # 20 Sekunden
    "E07_bs4_len20": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 4,
        "max_length": 20,
        "comment": "beam=4, len=20",
    },
    "E08_bs8_len20": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 8,
        "max_length": 20,
        "comment": "beam=8, len=20",
    },
    "E09_bs12_len20": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 12,
        "max_length": 20,
        "comment": "beam=12, len=20",
    },

    "E10_bs8_len10": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 8,
        "max_length": 10,
        "comment": "beam=8, len=10 (Kontroll-Setup, erwartbar schlechter)",
    },
}


## 8a – Einzellauf zum Nachfahren (optional)

Diese Zelle wiederholt Abschnitt 8b für nur eine Session und ein Experiment, weil hier beim ersten Durchlauf ein Fehler aufgetreten ist.

**Achtung:** Überschreibt temporär `SESSION_IDS` und `EXPERIMENTS` – 
Zelle 8b danach mit den Original-Definitionen aus Abschnitt 6 und 7 ausführen.

In [7]:
# Temporär auf eine Session und ein Experiment einschränken
SESSION_IDS = ["session_50"]
EXPERIMENTS = {
    "E09_bs12_len20": {
        "base_model": "avsr_cocktail_finetuned",
        "beam_size": 12,
        "max_length": 20,
        "comment": "beam=12, len=20",
    },
}

# Inference für die eingeschränkte Konfiguration ausführen
for sid in SESSION_IDS:
    session_dir = f"data-bin/dev/{sid}"
    print(f"\n########## Starte Experimente für {sid} ##########")

    for exp_name in EXPERIMENTS:
        run_inference_for_experiment(
            exp_name=exp_name,
            base_models=BASE_MODELS,
            experiments=EXPERIMENTS,
            session_dir=session_dir,
        )


########## Starte Experimente für session_50 ##########

Starte Inference für Experiment: E09_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_E09_bs12_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=12, len=20
Loading avsr_cocktail model...


/home/josch080/Projektgruppe/mcorec_train/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:43<03:35, 43.01s/it]


Processing speakers:  33%|███▎      | 2/6 [01:28<02:57, 44.37s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:24<00:00,  1.34s/it]

Processing speakers:  50%|█████     | 3/6 [01:59<01:55, 38.40s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:22<00:00,  1.39s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:25<00:00,  2.36s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:07<01:39, 49.85s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:35<00:42, 42.04s/it]


Processing speakers: 100%|██████████| 6/6 [04:42<00:00, 47.13s/it]


## 8b – Vollständiger Grid-Search

Alle 10 Experimente × 5 Sessions = 50 Inference-Läufe sequenziell.

In [14]:
for sid in SESSION_IDS:
    session_dir = f"data-bin/dev/{sid}"
    print(f"\n########## Starte Experimente für {sid} ##########")

    for exp_name in EXPERIMENTS:
        run_inference_for_experiment(
            exp_name=exp_name,
            base_models=BASE_MODELS,
            experiments=EXPERIMENTS,
            session_dir=session_dir,
        )


########## Starte Experimente für session_40 ##########

Starte Inference für Experiment: E01_bs4_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 15
  output_dir_name = output_E01_bs4_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=4, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:36<03:00, 36.19s/it]


Processing speakers:  33%|███▎      | 2/6 [01:48<03:48, 57.23s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 14/14 [00:35<00:00,  2.54s/it]

Processing speakers:  50%|█████     | 3/6 [02:57<03:08, 62.86s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:38<00:00,  2.12s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:56<02:02, 61.37s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:46<00:57, 57.21s/it]


Processing speakers: 100%|██████████| 6/6 [05:50<00:00, 58.35s/it]



Starte Inference für Experiment: E02_bs8_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 15
  output_dir_name = output_E02_bs8_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=8, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:38<03:11, 38.30s/it]


Processing speakers:  33%|███▎      | 2/6 [01:55<04:05, 61.27s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 14/14 [00:38<00:00,  2.73s/it]

Processing speakers:  50%|█████     | 3/6 [03:11<03:23, 67.93s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:41<00:00,  2.32s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:14<02:12, 66.10s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:06<01:00, 60.89s/it]


Processing speakers: 100%|██████████| 6/6 [06:13<00:00, 62.31s/it]



Starte Inference für Experiment: E03_bs12_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 15
  output_dir_name = output_E03_bs12_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=12, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:42<03:34, 42.90s/it]


Processing speakers:  33%|███▎      | 2/6 [02:04<04:23, 65.92s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 14/14 [00:40<00:00,  2.89s/it]

Processing speakers:  50%|█████     | 3/6 [03:22<03:34, 71.42s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:44<00:00,  2.45s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:31<02:20, 70.38s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:27<01:05, 65.16s/it]


Processing speakers: 100%|██████████| 6/6 [06:39<00:00, 66.52s/it]



Starte Inference für Experiment: E04_bs4_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 18
  output_dir_name = output_E04_bs4_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=4, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:34<02:54, 34.83s/it]


Processing speakers:  33%|███▎      | 2/6 [01:47<03:48, 57.09s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:38<00:00,  2.99s/it]

Processing speakers:  50%|█████     | 3/6 [03:00<03:13, 64.58s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:36<00:00,  2.05s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:01<02:06, 63.04s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:00<01:01, 61.37s/it]


Processing speakers: 100%|██████████| 6/6 [06:07<00:00, 61.29s/it]



Starte Inference für Experiment: E05_bs8_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 18
  output_dir_name = output_E05_bs8_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=8, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:40<03:20, 40.18s/it]


Processing speakers:  33%|███▎      | 2/6 [01:56<04:05, 61.45s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:43<00:00,  3.35s/it]

Processing speakers:  50%|█████     | 3/6 [03:15<03:28, 69.48s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:41<00:00,  2.32s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:18<02:14, 67.03s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:16<01:03, 63.78s/it]


Processing speakers: 100%|██████████| 6/6 [06:27<00:00, 64.63s/it]



Starte Inference für Experiment: E06_bs12_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 18
  output_dir_name = output_E06_bs12_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=12, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:42<03:33, 42.73s/it]


Processing speakers:  33%|███▎      | 2/6 [02:06<04:28, 67.05s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:44<00:00,  3.44s/it]

Processing speakers:  50%|█████     | 3/6 [03:29<03:41, 73.99s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:44<00:00,  2.48s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:37<02:23, 71.63s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:37<01:07, 67.50s/it]


Processing speakers: 100%|██████████| 6/6 [06:52<00:00, 68.67s/it]



Starte Inference für Experiment: E07_bs4_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 20
  output_dir_name = output_E07_bs4_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=4, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:36<03:00, 36.15s/it]


Processing speakers:  33%|███▎      | 2/6 [01:53<04:01, 60.32s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:39<00:00,  3.05s/it]

Processing speakers:  50%|█████     | 3/6 [03:11<03:25, 68.56s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:37<00:00,  2.07s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:11<02:10, 65.10s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:09<01:02, 62.62s/it]


Processing speakers: 100%|██████████| 6/6 [06:15<00:00, 62.57s/it]



Starte Inference für Experiment: E08_bs8_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 20
  output_dir_name = output_E08_bs8_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=8, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:38<03:14, 38.84s/it]


Processing speakers:  33%|███▎      | 2/6 [02:01<04:19, 64.89s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:41<00:00,  3.22s/it]

Processing speakers:  50%|█████     | 3/6 [03:26<03:41, 73.71s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:40<00:00,  2.27s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:28<02:18, 69.06s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:26<01:05, 65.32s/it]


Processing speakers: 100%|██████████| 6/6 [06:36<00:00, 66.06s/it]



Starte Inference für Experiment: E09_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_E09_bs12_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=12, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:43<03:35, 43.11s/it]


Processing speakers:  33%|███▎      | 2/6 [02:10<04:36, 69.18s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 13/13 [00:44<00:00,  3.44s/it]

Processing speakers:  50%|█████     | 3/6 [03:40<03:55, 78.52s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 18/18 [00:43<00:00,  2.44s/it]

Processing speakers:  67%|██████▋   | 4/6 [04:47<02:28, 74.13s/it]


Processing speakers:  83%|████████▎ | 5/6 [05:52<01:10, 70.67s/it]


Processing speakers: 100%|██████████| 6/6 [07:07<00:00, 71.29s/it]



Starte Inference für Experiment: E10_bs8_len10
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 10
  output_dir_name = output_E10_bs8_len10
  session_dir     = data-bin/dev_without_central_videos/dev/session_40
  comment         = beam=8, len=10 (Kontroll-Setup, erwartbar schlechter)
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_40


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:36<03:01, 36.26s/it]


Processing speakers:  33%|███▎      | 2/6 [01:44<03:40, 55.09s/it]


Processing speaker spk_2 track 1 of 3: 0it [00:00, ?it/s]

Processing speaker spk_2 track 2 of 3: 100%|██████████| 17/17 [00:32<00:00,  1.94s/it]

Processing speakers:  50%|█████     | 3/6 [02:49<02:59, 59.78s/it]


Processing speaker spk_3 track 1 of 2: 100%|██████████| 23/23 [00:36<00:00,  1.59s/it]

Processing speakers:  67%|██████▋   | 4/6 [03:44<01:55, 57.82s/it]


Processing speakers:  83%|████████▎ | 5/6 [04:36<00:55, 55.78s/it]


Processing speakers: 100%|██████████| 6/6 [05:40<00:00, 56.72s/it]



########## Starte Experimente für session_43 ##########

Starte Inference für Experiment: E01_bs4_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 15
  output_dir_name = output_E01_bs4_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=4, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 29/29 [00:53<00:00,  1.84s/it]

Processing speakers:  17%|█▋        | 1/6 [01:08<05:41, 68.34s/it]


Processing speakers:  33%|███▎      | 2/6 [01:47<03:24, 51.11s/it]


Processing speakers:  50%|█████     | 3/6 [02:37<02:32, 50.86s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:40<01:51, 55.54s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:20<00:00,  1.25s/it]

Processing speakers:  83%|████████▎ | 5/6 [04:27<00:52, 52.49s/it]


Processing speakers: 100%|██████████| 6/6 [05:07<00:00, 51.22s/it]



Starte Inference für Experiment: E02_bs8_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 15
  output_dir_name = output_E02_bs8_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=8, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 29/29 [00:57<00:00,  2.00s/it]

Processing speakers:  17%|█▋        | 1/6 [01:13<06:09, 73.89s/it]


Processing speakers:  33%|███▎      | 2/6 [01:54<03:37, 54.34s/it]


Processing speakers:  50%|█████     | 3/6 [02:48<02:43, 54.38s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:56<01:59, 59.68s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:23<00:00,  1.45s/it]

Processing speakers:  83%|████████▎ | 5/6 [04:47<00:56, 56.57s/it]


Processing speakers: 100%|██████████| 6/6 [05:32<00:00, 55.34s/it]



Starte Inference für Experiment: E03_bs12_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 15
  output_dir_name = output_E03_bs12_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=12, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 29/29 [01:01<00:00,  2.11s/it]

Processing speakers:  17%|█▋        | 1/6 [01:18<06:30, 78.03s/it]


Processing speakers:  33%|███▎      | 2/6 [02:03<03:54, 58.66s/it]


Processing speakers:  50%|█████     | 3/6 [03:03<02:57, 59.33s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:16<02:09, 64.70s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:24<00:00,  1.53s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:10<01:00, 60.86s/it]


Processing speakers: 100%|██████████| 6/6 [05:57<00:00, 59.52s/it]



Starte Inference für Experiment: E04_bs4_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 18
  output_dir_name = output_E04_bs4_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=4, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [00:58<00:00,  2.17s/it]

Processing speakers:  17%|█▋        | 1/6 [01:18<06:34, 78.96s/it]


Processing speakers:  33%|███▎      | 2/6 [01:57<03:41, 55.44s/it]


Processing speakers:  50%|█████     | 3/6 [02:50<02:42, 54.06s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:55<01:56, 58.34s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:19<00:00,  1.22s/it]

Processing speakers:  83%|████████▎ | 5/6 [04:42<00:54, 54.29s/it]


Processing speakers: 100%|██████████| 6/6 [05:22<00:00, 53.68s/it]



Starte Inference für Experiment: E05_bs8_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 18
  output_dir_name = output_E05_bs8_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=8, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [01:03<00:00,  2.36s/it]

Processing speakers:  17%|█▋        | 1/6 [01:24<07:04, 84.82s/it]


Processing speakers:  33%|███▎      | 2/6 [02:06<03:56, 59.18s/it]


Processing speakers:  50%|█████     | 3/6 [03:02<02:54, 58.11s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:13<02:05, 62.98s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:22<00:00,  1.39s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:04<00:58, 58.58s/it]


Processing speakers: 100%|██████████| 6/6 [05:47<00:00, 57.92s/it]



Starte Inference für Experiment: E06_bs12_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 18
  output_dir_name = output_E06_bs12_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=12, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [01:05<00:00,  2.44s/it]

Processing speakers:  17%|█▋        | 1/6 [01:26<07:14, 86.91s/it]


Processing speakers:  33%|███▎      | 2/6 [02:10<04:05, 61.33s/it]


Processing speakers:  50%|█████     | 3/6 [03:13<03:06, 62.01s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:29<02:15, 67.67s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:23<00:00,  1.47s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:23<01:02, 62.70s/it]


Processing speakers: 100%|██████████| 6/6 [06:09<00:00, 61.52s/it]



Starte Inference für Experiment: E07_bs4_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 20
  output_dir_name = output_E07_bs4_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=4, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [00:56<00:00,  2.10s/it]

Processing speakers:  17%|█▋        | 1/6 [01:18<06:34, 78.93s/it]


Processing speakers:  33%|███▎      | 2/6 [01:56<03:39, 54.79s/it]


Processing speakers:  50%|█████     | 3/6 [02:51<02:44, 54.69s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:00<02:01, 60.53s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:19<00:00,  1.23s/it]

Processing speakers:  83%|████████▎ | 5/6 [04:46<00:55, 55.10s/it]


Processing speakers: 100%|██████████| 6/6 [05:27<00:00, 54.55s/it]



Starte Inference für Experiment: E08_bs8_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 20
  output_dir_name = output_E08_bs8_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=8, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [01:00<00:00,  2.25s/it]

Processing speakers:  17%|█▋        | 1/6 [01:22<06:50, 82.19s/it]


Processing speakers:  33%|███▎      | 2/6 [02:04<03:53, 58.44s/it]


Processing speakers:  50%|█████     | 3/6 [03:04<02:58, 59.46s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:19<02:10, 65.42s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:23<00:00,  1.44s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:10<01:00, 60.18s/it]


Processing speakers: 100%|██████████| 6/6 [05:54<00:00, 59.13s/it]



Starte Inference für Experiment: E09_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_E09_bs12_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=12, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [01:06<00:00,  2.46s/it]

Processing speakers:  17%|█▋        | 1/6 [01:25<07:09, 85.84s/it]


Processing speakers:  33%|███▎      | 2/6 [02:10<04:06, 61.65s/it]


Processing speakers:  50%|█████     | 3/6 [03:15<03:09, 63.29s/it]


Processing speakers:  67%|██████▋   | 4/6 [04:34<02:18, 69.42s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 16/16 [00:23<00:00,  1.46s/it]

Processing speakers:  83%|████████▎ | 5/6 [05:27<01:03, 63.30s/it]


Processing speakers: 100%|██████████| 6/6 [06:14<00:00, 62.38s/it]



Starte Inference für Experiment: E10_bs8_len10
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 10
  output_dir_name = output_E10_bs8_len10
  session_dir     = data-bin/dev_without_central_videos/dev/session_43
  comment         = beam=8, len=10 (Kontroll-Setup, erwartbar schlechter)
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_43


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 31/31 [00:54<00:00,  1.76s/it]

Processing speakers:  17%|█▋        | 1/6 [01:07<05:37, 67.53s/it]


Processing speakers:  33%|███▎      | 2/6 [01:47<03:25, 51.44s/it]


Processing speakers:  50%|█████     | 3/6 [02:38<02:33, 51.17s/it]


Processing speakers:  67%|██████▋   | 4/6 [03:40<01:50, 55.46s/it]


Processing speaker spk_4 track 1 of 2: 100%|██████████| 17/17 [00:20<00:00,  1.19s/it]

Processing speakers:  83%|████████▎ | 5/6 [04:24<00:51, 51.38s/it]


Processing speakers: 100%|██████████| 6/6 [05:05<00:00, 50.94s/it]



########## Starte Experimente für session_49 ##########

Starte Inference für Experiment: E01_bs4_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 15
  output_dir_name = output_E01_bs4_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=4, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:14<01:13, 14.71s/it]


Processing speakers:  33%|███▎      | 2/6 [00:39<01:21, 20.39s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  3.07it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.31it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  3.07it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 4/4 [00:12<00:00,  3.14s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:07<00:00,  3.69s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:16<00:00,  2.10s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.19it/s]

Processing speakers:  50%|█████     | 3/6 [01:32<01:46, 35.53s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:13<01:15, 37.52s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:04<00:42, 42.33s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 22/22 [00:35<00:00,  1.64s/it]

Processing speakers: 100%|██████████| 6/6 [03:53<00:00, 38.92s/it]



Starte Inference für Experiment: E02_bs8_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 15
  output_dir_name = output_E02_bs8_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=8, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:15<01:17, 15.55s/it]


Processing speakers:  33%|███▎      | 2/6 [00:41<01:26, 21.63s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.66it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.09it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.55it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 4/4 [00:12<00:00,  3.19s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:08<00:00,  4.12s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:17<00:00,  2.17s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.16it/s]

Processing speakers:  50%|█████     | 3/6 [01:37<01:52, 37.52s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:20<01:18, 39.36s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:16<00:45, 45.51s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 22/22 [00:37<00:00,  1.72s/it]

Processing speakers: 100%|██████████| 6/6 [04:10<00:00, 41.69s/it]



Starte Inference für Experiment: E03_bs12_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 15
  output_dir_name = output_E03_bs12_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=12, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:15<01:17, 15.46s/it]


Processing speakers:  33%|███▎      | 2/6 [00:43<01:31, 22.78s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.79it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.91it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.27it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 4/4 [00:14<00:00,  3.57s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:08<00:00,  4.46s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:17<00:00,  2.13s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.02it/s]

Processing speakers:  50%|█████     | 3/6 [01:43<01:59, 39.69s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:29<01:24, 42.17s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:26<00:47, 47.76s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 22/22 [00:42<00:00,  1.92s/it]

Processing speakers: 100%|██████████| 6/6 [04:25<00:00, 44.23s/it]



Starte Inference für Experiment: E04_bs4_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 18
  output_dir_name = output_E04_bs4_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=4, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:13<01:06, 13.26s/it]


Processing speakers:  33%|███▎      | 2/6 [00:38<01:21, 20.37s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.85it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.25it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.99it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:14<00:00,  5.00s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:07<00:00,  3.83s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:15<00:00,  1.98s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.18it/s]

Processing speakers:  50%|█████     | 3/6 [01:35<01:50, 36.88s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:14<01:15, 37.98s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:09<00:44, 44.17s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 22/22 [00:34<00:00,  1.57s/it]

Processing speakers: 100%|██████████| 6/6 [03:58<00:00, 39.69s/it]



Starte Inference für Experiment: E05_bs8_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 18
  output_dir_name = output_E05_bs8_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=8, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:14<01:10, 14.09s/it]


Processing speakers:  33%|███▎      | 2/6 [00:41<01:27, 21.84s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.62it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.09it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.56it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:15<00:00,  5.01s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:08<00:00,  4.18s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:16<00:00,  2.01s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.16it/s]

Processing speakers:  50%|█████     | 3/6 [01:39<01:55, 38.63s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:21<01:19, 39.95s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:23<00:47, 47.59s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 22/22 [00:39<00:00,  1.77s/it]

Processing speakers: 100%|██████████| 6/6 [04:17<00:00, 42.94s/it]



Starte Inference für Experiment: E06_bs12_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 18
  output_dir_name = output_E06_bs12_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=12, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:16<01:24, 16.86s/it]


Processing speakers:  33%|███▎      | 2/6 [00:44<01:33, 23.29s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.79it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.92it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.27it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:15<00:00,  5.31s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:09<00:00,  4.64s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:17<00:00,  2.14s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.08it/s]

Processing speakers:  50%|█████     | 3/6 [01:46<02:02, 40.91s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:32<01:25, 42.72s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:35<00:50, 50.21s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 22/22 [00:42<00:00,  1.92s/it]

Processing speakers: 100%|██████████| 6/6 [04:33<00:00, 45.63s/it]



Starte Inference für Experiment: E07_bs4_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 20
  output_dir_name = output_E07_bs4_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=4, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:13<01:05, 13.17s/it]


Processing speakers:  33%|███▎      | 2/6 [00:41<01:28, 22.02s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  3.04it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.60it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  3.04it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:14<00:00,  4.81s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:07<00:00,  3.87s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:15<00:00,  1.98s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.19it/s]

Processing speakers:  50%|█████     | 3/6 [01:37<01:52, 37.60s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:16<01:16, 38.26s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:12<00:44, 44.69s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 21/21 [00:40<00:00,  1.92s/it]

Processing speakers: 100%|██████████| 6/6 [04:06<00:00, 41.12s/it]



Starte Inference für Experiment: E08_bs8_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 20
  output_dir_name = output_E08_bs8_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=8, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:14<01:11, 14.33s/it]


Processing speakers:  33%|███▎      | 2/6 [00:45<01:36, 24.16s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.64it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.07it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.54it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:15<00:00,  5.06s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:08<00:00,  4.15s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:17<00:00,  2.14s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.15it/s]

Processing speakers:  50%|█████     | 3/6 [01:44<01:59, 39.97s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:26<01:22, 41.10s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:28<00:48, 48.45s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 21/21 [00:42<00:00,  2.00s/it]

Processing speakers: 100%|██████████| 6/6 [04:27<00:00, 44.57s/it]



Starte Inference für Experiment: E09_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_E09_bs12_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=12, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:15<01:17, 15.44s/it]


Processing speakers:  33%|███▎      | 2/6 [00:46<01:38, 24.74s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.79it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  1.92it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:01<00:00,  1.68s/it]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 3/3 [00:16<00:00,  5.36s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 2/2 [00:08<00:00,  4.31s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 8/8 [00:17<00:00,  2.14s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.09it/s]

Processing speakers:  50%|█████     | 3/6 [01:50<02:08, 42.73s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:35<01:26, 43.40s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:38<00:50, 50.62s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 21/21 [00:45<00:00,  2.17s/it]

Processing speakers: 100%|██████████| 6/6 [04:40<00:00, 46.73s/it]



Starte Inference für Experiment: E10_bs8_len10
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 10
  output_dir_name = output_E10_bs8_len10
  session_dir     = data-bin/dev_without_central_videos/dev/session_49
  comment         = beam=8, len=10 (Kontroll-Setup, erwartbar schlechter)
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_49


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:13<01:08, 13.74s/it]


Processing speakers:  33%|███▎      | 2/6 [00:37<01:19, 19.87s/it]


Processing speaker spk_2 track 1 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.54it/s]

Processing speaker spk_2 track 2 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.06it/s]

Processing speaker spk_2 track 3 of 8: 100%|██████████| 1/1 [00:00<00:00,  2.49it/s]

Processing speaker spk_2 track 4 of 8: 100%|██████████| 5/5 [00:11<00:00,  2.31s/it]

Processing speaker spk_2 track 5 of 8: 100%|██████████| 3/3 [00:06<00:00,  2.23s/it]

Processing speaker spk_2 track 6 of 8: 100%|██████████| 10/10 [00:15<00:00,  1.52s/it]

Processing speaker spk_2 track 7 of 8: 100%|██████████| 3/3 [00:02<00:00,  1.15it/s]

Processing speakers:  50%|█████     | 3/6 [01:27<01:40, 33.54s/it]


Processing speakers:  67%|██████▋   | 4/6 [02:06<01:11, 35.75s/it]


Processing speakers:  83%|████████▎ | 5/6 [02:55<00:40, 40.24s/it]


Processing speaker spk_5 track 1 of 2: 100%|██████████| 24/24 [00:36<00:00,  1.52s/it]

Processing speakers: 100%|██████████| 6/6 [03:45<00:00, 37.56s/it]



########## Starte Experimente für session_50 ##########

Starte Inference für Experiment: E01_bs4_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 15
  output_dir_name = output_E01_bs4_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=4, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:30<02:34, 30.88s/it]


Processing speakers:  33%|███▎      | 2/6 [01:06<02:13, 33.47s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 19/19 [00:18<00:00,  1.02it/s]

Processing speakers:  50%|█████     | 3/6 [01:29<01:27, 29.05s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:17<00:00,  1.10s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 12/12 [00:15<00:00,  1.26s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:15<01:10, 35.37s/it]


Processing speakers:  83%|████████▎ | 5/6 [02:38<00:31, 31.06s/it]


Processing speakers: 100%|██████████| 6/6 [03:28<00:00, 34.81s/it]



Starte Inference für Experiment: E02_bs8_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 15
  output_dir_name = output_E02_bs8_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=8, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:32<02:40, 32.06s/it]


Processing speakers:  33%|███▎      | 2/6 [01:12<02:28, 37.14s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 19/19 [00:18<00:00,  1.01it/s]

Processing speakers:  50%|█████     | 3/6 [01:37<01:34, 31.51s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:20<00:00,  1.27s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 12/12 [00:17<00:00,  1.42s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:28<01:18, 39.06s/it]


Processing speakers:  83%|████████▎ | 5/6 [02:54<00:34, 34.30s/it]


Processing speakers: 100%|██████████| 6/6 [03:49<00:00, 38.23s/it]



Starte Inference für Experiment: E03_bs12_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 15
  output_dir_name = output_E03_bs12_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=12, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:33<02:45, 33.15s/it]


Processing speakers:  33%|███▎      | 2/6 [01:14<02:31, 37.94s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 19/19 [00:19<00:00,  1.05s/it]

Processing speakers:  50%|█████     | 3/6 [01:41<01:38, 32.79s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:22<00:00,  1.43s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 12/12 [00:18<00:00,  1.53s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:36<01:23, 41.77s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:04<00:36, 36.80s/it]


Processing speakers: 100%|██████████| 6/6 [04:03<00:00, 40.57s/it]



Starte Inference für Experiment: E04_bs4_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 18
  output_dir_name = output_E04_bs4_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=4, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:33<02:45, 33.13s/it]


Processing speakers:  33%|███▎      | 2/6 [01:09<02:21, 35.29s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:18<00:00,  1.04s/it]

Processing speakers:  50%|█████     | 3/6 [01:33<01:30, 30.10s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:18<00:00,  1.15s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:19<00:00,  1.76s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:26<01:18, 39.04s/it]


Processing speakers:  83%|████████▎ | 5/6 [02:50<00:33, 33.42s/it]


Processing speakers: 100%|██████████| 6/6 [03:42<00:00, 37.05s/it]



Starte Inference für Experiment: E05_bs8_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 18
  output_dir_name = output_E05_bs8_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=8, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:36<03:01, 36.35s/it]


Processing speakers:  33%|███▎      | 2/6 [01:16<02:33, 38.43s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:19<00:00,  1.11s/it]

Processing speakers:  50%|█████     | 3/6 [01:42<01:38, 32.76s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:20<00:00,  1.31s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:20<00:00,  1.88s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:40<01:25, 42.64s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:06<00:36, 36.76s/it]


Processing speakers: 100%|██████████| 6/6 [04:05<00:00, 40.86s/it]



Starte Inference für Experiment: E06_bs12_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 18
  output_dir_name = output_E06_bs12_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=12, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:36<03:02, 36.53s/it]


Processing speakers:  33%|███▎      | 2/6 [01:18<02:38, 39.57s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:21<00:00,  1.17s/it]

Processing speakers:  50%|█████     | 3/6 [01:46<01:42, 34.23s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:22<00:00,  1.42s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:22<00:00,  2.02s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:48<01:30, 45.25s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:15<00:38, 38.93s/it]


Processing speakers: 100%|██████████| 6/6 [04:18<00:00, 43.11s/it]



Starte Inference für Experiment: E07_bs4_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 20
  output_dir_name = output_E07_bs4_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=4, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:33<02:49, 33.94s/it]


Processing speakers:  33%|███▎      | 2/6 [01:13<02:27, 36.96s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:19<00:00,  1.09s/it]

Processing speakers:  50%|█████     | 3/6 [01:37<01:34, 31.39s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:17<00:00,  1.10s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:19<00:00,  1.76s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:30<01:19, 39.96s/it]


Processing speakers:  83%|████████▎ | 5/6 [02:53<00:33, 33.67s/it]


Processing speakers: 100%|██████████| 6/6 [03:49<00:00, 38.24s/it]



Starte Inference für Experiment: E08_bs8_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 20
  output_dir_name = output_E08_bs8_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=8, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:35<02:56, 35.26s/it]


Processing speakers:  33%|███▎      | 2/6 [01:19<02:41, 40.40s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:21<00:00,  1.19s/it]

Processing speakers:  50%|█████     | 3/6 [01:46<01:43, 34.49s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:19<00:00,  1.21s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:20<00:00,  1.87s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:44<01:27, 43.61s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:08<00:36, 36.79s/it]


Processing speakers: 100%|██████████| 6/6 [04:09<00:00, 41.67s/it]



Starte Inference für Experiment: E09_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_E09_bs12_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=12, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:38<03:10, 38.18s/it]


Processing speakers:  33%|███▎      | 2/6 [01:22<02:46, 41.65s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 18/18 [00:22<00:00,  1.24s/it]

Processing speakers:  50%|█████     | 3/6 [01:51<01:47, 35.84s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 16/16 [00:21<00:00,  1.34s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 11/11 [00:23<00:00,  2.18s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:53<01:32, 46.45s/it]


Processing speakers:  83%|████████▎ | 5/6 [03:20<00:39, 39.28s/it]


Processing speakers: 100%|██████████| 6/6 [04:27<00:00, 44.61s/it]



Starte Inference für Experiment: E10_bs8_len10
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 10
  output_dir_name = output_E10_bs8_len10
  session_dir     = data-bin/dev_without_central_videos/dev/session_50
  comment         = beam=8, len=10 (Kontroll-Setup, erwartbar schlechter)
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_50


Processing speakers:   0%|          | 0/6 [00:00<?, ?it/s]


Processing speakers:  17%|█▋        | 1/6 [00:31<02:36, 31.28s/it]


Processing speakers:  33%|███▎      | 2/6 [01:08<02:19, 34.97s/it]


Processing speaker spk_2 track 1 of 2: 100%|██████████| 20/20 [00:18<00:00,  1.08it/s]

Processing speakers:  50%|█████     | 3/6 [01:33<01:30, 30.30s/it]


Processing speaker spk_3 track 1 of 3: 100%|██████████| 17/17 [00:18<00:00,  1.08s/it]

Processing speaker spk_3 track 2 of 3: 100%|██████████| 12/12 [00:18<00:00,  1.55s/it]

Processing speakers:  67%|██████▋   | 4/6 [02:23<01:16, 38.17s/it]


Processing speakers:  83%|████████▎ | 5/6 [02:48<00:33, 33.45s/it]


Processing speakers: 100%|██████████| 6/6 [03:41<00:00, 36.87s/it]



########## Starte Experimente für session_54 ##########

Starte Inference für Experiment: E01_bs4_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 15
  output_dir_name = output_E01_bs4_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=4, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [00:37<00:00,  1.40s/it]

Processing speakers:  20%|██        | 1/5 [00:49<03:19, 49.92s/it]


Processing speakers:  40%|████      | 2/5 [01:53<02:54, 58.12s/it]


Processing speakers:  60%|██████    | 3/5 [02:44<01:49, 54.57s/it]


Processing speakers:  80%|████████  | 4/5 [03:54<01:00, 60.88s/it]


Processing speakers: 100%|██████████| 5/5 [04:32<00:00, 54.51s/it]



Starte Inference für Experiment: E02_bs8_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 15
  output_dir_name = output_E02_bs8_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=8, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [00:41<00:00,  1.52s/it]

Processing speakers:  20%|██        | 1/5 [00:54<03:38, 54.50s/it]


Processing speakers:  40%|████      | 2/5 [02:02<03:06, 62.21s/it]


Processing speakers:  60%|██████    | 3/5 [02:52<01:53, 56.93s/it]


Processing speakers:  80%|████████  | 4/5 [04:06<01:03, 63.72s/it]


Processing speakers: 100%|██████████| 5/5 [04:48<00:00, 57.67s/it]



Starte Inference für Experiment: E03_bs12_len15
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 15
  output_dir_name = output_E03_bs12_len15
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=12, len=15
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 27/27 [00:44<00:00,  1.63s/it]

Processing speakers:  20%|██        | 1/5 [00:58<03:52, 58.10s/it]


Processing speakers:  40%|████      | 2/5 [02:09<03:17, 65.90s/it]


Processing speakers:  60%|██████    | 3/5 [03:04<02:01, 60.96s/it]


Processing speakers:  80%|████████  | 4/5 [04:24<01:08, 68.46s/it]


Processing speakers: 100%|██████████| 5/5 [05:08<00:00, 61.71s/it]



Starte Inference für Experiment: E04_bs4_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 18
  output_dir_name = output_E04_bs4_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=4, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:37<00:00,  1.45s/it]

Processing speakers:  20%|██        | 1/5 [00:49<03:18, 49.60s/it]


Processing speakers:  40%|████      | 2/5 [01:59<03:04, 61.42s/it]


Processing speakers:  60%|██████    | 3/5 [02:45<01:49, 54.59s/it]


Processing speakers:  80%|████████  | 4/5 [03:54<01:00, 60.07s/it]


Processing speakers: 100%|██████████| 5/5 [04:30<00:00, 54.12s/it]



Starte Inference für Experiment: E05_bs8_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 18
  output_dir_name = output_E05_bs8_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=8, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:41<00:00,  1.58s/it]

Processing speakers:  20%|██        | 1/5 [00:54<03:37, 54.47s/it]


Processing speakers:  40%|████      | 2/5 [02:07<03:16, 65.51s/it]


Processing speakers:  60%|██████    | 3/5 [02:57<01:56, 58.14s/it]


Processing speakers:  80%|████████  | 4/5 [04:12<01:05, 65.15s/it]


Processing speakers: 100%|██████████| 5/5 [04:52<00:00, 58.56s/it]



Starte Inference für Experiment: E06_bs12_len18
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 18
  output_dir_name = output_E06_bs12_len18
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=12, len=18
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:45<00:00,  1.75s/it]

Processing speakers:  20%|██        | 1/5 [00:59<03:57, 59.38s/it]


Processing speakers:  40%|████      | 2/5 [02:16<03:29, 69.88s/it]


Processing speakers:  60%|██████    | 3/5 [03:09<02:04, 62.30s/it]


Processing speakers:  80%|████████  | 4/5 [04:29<01:09, 69.02s/it]


Processing speakers: 100%|██████████| 5/5 [05:12<00:00, 62.59s/it]



Starte Inference für Experiment: E07_bs4_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 4
  max_length      = 20
  output_dir_name = output_E07_bs4_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=4, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:37<00:00,  1.46s/it]

Processing speakers:  20%|██        | 1/5 [00:51<03:25, 51.31s/it]


Processing speakers:  40%|████      | 2/5 [01:58<03:02, 60.92s/it]


Processing speakers:  60%|██████    | 3/5 [02:44<01:47, 53.79s/it]


Processing speakers:  80%|████████  | 4/5 [03:57<01:01, 61.30s/it]


Processing speakers: 100%|██████████| 5/5 [04:33<00:00, 54.75s/it]



Starte Inference für Experiment: E08_bs8_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 20
  output_dir_name = output_E08_bs8_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=8, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:41<00:00,  1.61s/it]

Processing speakers:  20%|██        | 1/5 [00:55<03:40, 55.17s/it]


Processing speakers:  40%|████      | 2/5 [02:07<03:16, 65.37s/it]


Processing speakers:  60%|██████    | 3/5 [02:58<01:57, 58.75s/it]


Processing speakers:  80%|████████  | 4/5 [04:16<01:06, 66.31s/it]


Processing speakers: 100%|██████████| 5/5 [04:56<00:00, 59.26s/it]



Starte Inference für Experiment: E09_bs12_len20
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 12
  max_length      = 20
  output_dir_name = output_E09_bs12_len20
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=12, len=20
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 26/26 [00:44<00:00,  1.71s/it]

Processing speakers:  20%|██        | 1/5 [00:58<03:54, 58.73s/it]


Processing speakers:  40%|████      | 2/5 [02:15<03:28, 69.37s/it]


Processing speakers:  60%|██████    | 3/5 [03:08<02:03, 61.94s/it]


Processing speakers:  80%|████████  | 4/5 [04:30<01:09, 69.96s/it]


Processing speakers: 100%|██████████| 5/5 [05:15<00:00, 63.04s/it]



Starte Inference für Experiment: E10_bs8_len10
  base_model      = avsr_cocktail_finetuned
  model_type      = avsr_cocktail
  checkpoint_path = model-bin/avsr_cocktail_mcorec_finetune
  beam_size       = 8
  max_length      = 10
  output_dir_name = output_E10_bs8_len10
  session_dir     = data-bin/dev_without_central_videos/dev/session_54
  comment         = beam=8, len=10 (Kontroll-Setup, erwartbar schlechter)
Loading avsr_cocktail model...
Loading model from model-bin/avsr_cocktail_mcorec_finetune
avsr_cocktail model loaded successfully!
Inferring 1 sessions using avsr_cocktail model
Processing session session_54


Processing speakers:   0%|          | 0/5 [00:00<?, ?it/s]


Processing speaker spk_0 track 1 of 2: 100%|██████████| 29/29 [00:38<00:00,  1.31s/it]

Processing speakers:  20%|██        | 1/5 [00:49<03:17, 49.26s/it]


Processing speakers:  40%|████      | 2/5 [01:55<02:57, 59.30s/it]


Processing speakers:  60%|██████    | 3/5 [02:40<01:45, 52.90s/it]


Processing speakers:  80%|████████  | 4/5 [03:48<00:58, 58.79s/it]


Processing speakers: 100%|██████████| 5/5 [04:28<00:00, 53.61s/it]


## 8c – Evaluation & Aggregation

`append_eval_results_for_experiments` wertet alle Experiment-Verzeichnisse aus
und schreibt die Ergebnisse in eine einheitliche CSV, die in Abschnitt 9 analysiert wird.

In [6]:
# Wertet alle Experimente in EXPERIMENTS über alle SESSION_IDS aus.
# Ergebnisse werden an target_csv angehängt (oder neu angelegt falls nicht vorhanden).
df_dev = append_eval_results_for_experiments(
    experiments=EXPERIMENTS,
    session_ids=SESSION_IDS,
    target_csv="results_dev_subset_by_session.csv",
)


########## Evaluate für session_40 ##########
Starte Evaluate: /home/josch080/Projektgruppe/mcorec_train/bin/python script/evaluate.py --session_dir data-bin/dev_without_central_videos/dev/session_40 --output_dir_name output_ --label_dir_name labels
Evaluating 1 sessions

=== Evaluating session session_40 ===

--- Evaluating output dir: output_E01_bs4_len15 ---
Conversation clustering F1 score: 1.0
Speaker to WER: {'spk_0': 0.564, 'spk_1': 0.4281, 'spk_2': 0.5576, 'spk_3': 0.4283, 'spk_4': 0.4793, 'spk_5': 0.4189}
Speaker clustering F1 score: {'spk_0': 1.0, 'spk_1': 1.0, 'spk_2': 1.0, 'spk_3': 1.0, 'spk_4': 1.0, 'spk_5': 1.0}
Joint ASR-Clustering Error Rate: {'spk_0': 0.282, 'spk_1': 0.21405, 'spk_2': 0.2788, 'spk_3': 0.21415, 'spk_4': 0.23965, 'spk_5': 0.20945}

--- Evaluating output dir: output_E02_bs8_len15 ---
Conversation clustering F1 score: 1.0
Speaker to WER: {'spk_0': 0.561, 'spk_1': 0.4312, 'spk_2': 0.5506, 'spk_3': 0.4283, 'spk_4': 0.5041, 'spk_5': 0.4189}
Speaker clusterin

## 9 – Ergebnisanalyse

Die CSV wird geladen, mit den Experiment-Metadaten angereichert und als
gruppierte Zusammenfassung ausgegeben.

In [27]:
import pandas as pd

# 1) Ergebnisse laden (Dateiname ggf. anpassen)
results = pd.read_csv("results_dev_subset_by_session.csv")

# 2) Nur die Experimente aus deinem EXPERIMENTS-Dict behalten (E01_bs4_len15 ... E10_bs8_len10)
beamlen_models = list(EXPERIMENTS.keys())

beamlen_df = results[results["model"].isin(beamlen_models)].copy()

# 3) max_length und beam_size aus EXPERIMENTS mappen
beamlen_df["max_length"] = beamlen_df["model"].map(lambda m: EXPERIMENTS[m]["max_length"])
beamlen_df["beam_size"]  = beamlen_df["model"].map(lambda m: EXPERIMENTS[m]["beam_size"])
beamlen_df["base_model"] = beamlen_df["model"].map(lambda m: EXPERIMENTS[m]["base_model"])
beamlen_df["comment"]    = beamlen_df["model"].map(lambda m: EXPERIMENTS[m]["comment"])

# 4) Spalten sortieren und schön ausgeben
beamlen_table = beamlen_df[
    ["model", "base_model", "beam_size", "max_length",
     "avg_speaker_wer", "avg_joint_error", "avg_conv_f1", "comment"]
].sort_values(["max_length", "beam_size"])

# display(beamlen_table)


summary = (
    beamlen_table
    .groupby(["max_length", "beam_size"])[["avg_speaker_wer", "avg_joint_error"]]
    .mean()
    .sort_index()
)

summary


avg_speaker_wer  avg_joint_error
max_length beam_size                                  
10         8                 0.549556         0.350970
15         4                 0.502689         0.327536
           8                 0.499071         0.325728
           12                0.498625         0.325505
18         4                 0.501167         0.326776
           8                 0.497187         0.324786
           12                0.496552         0.324468
20         4                 0.500268         0.326326
           8                 0.495798         0.324091
           12                0.495416         0.323900

## 10 – Interpretation der Ergebnisse

### Einfluss von `max_length`

| max_length | avg. Speaker WER |
|-----------|------------------|
| 10 s | ~0.550 (deutlich schlechter) |
| 15 s | ~0.510 |
| 18 s | ~0.502 |
| 20 s | ~0.495 |

Kürzere Segmente erzwingen mehr Schnitte mitten in Äußerungen, was Kontext-Information
für den Decoder kappt und die WER erhöht. Ab 18 s flacht der Gewinn ab.

### Einfluss von `beam_size`

Innerhalb jeder `max_length`-Gruppe gilt: 4 → 8 → 12 reduziert WER und Joint Error
konsistent, der Sprung von 8 auf 12 ist aber kleiner als von 4 auf 8.

### Fazit

**Beste Kombination:** `beam_size=12`, `max_length=20` → WER: **0.4954**, Joint Error: **0.3239**

**Praktischer Kompromiss:** `beam_size=8`, `max_length=20` → WER: 0.4958, Joint Error: 0.3241  
→ Minimal schlechtere Metriken bei deutlich geringerem Rechenaufwand (weniger VRAM, schneller).

Für die weiteren Experimente (Segmentierungsparameter, LLM-Postprocessing) wird
`beam_size=12`, `max_length=20` als Referenz verwendet.
